### ISSUE TITLE
Require relative "../foo" also matches "./foo"

### ISSUE THREAD

**Issue by straight-shoota** [crystal-lang/crystal#4742]
File `foo.cr`:
```crystal
require "./bar"     # => fine
require "./bar.cr"  # => fine
require "../bar"    # => does not fail, but should
require "../bar.cr" # => fails
```
File `bar.cr`:
```
{% puts "hey from bar.cr" %}
```

Running `crystal foo.cr`, all but the last line require the file `./bar.cr` unless `../bar.cr` exists. Requirering `../bar` should not match a file of that name in the same directory but create an error as `../bar.cr` does.

**Comment by MakeNowJust** [comment]
Default `CRYSTAL_CONFIG_PATH` contains `./lib` and the compiler finds a file from `CRYSTAL_CONFIG_PATH` even though the filename is relative (dot-started), so `require "../bar"` is resolved to `./lib/../bar` internally. I don't know why relative path finding is fallen back to `CRYSTAL_CONFIG_PATH` (and when is it useful?)

**Comment by straight-shoota** [comment]
Yes, that seems to be the case. If you put the example files into a subdirectory and run it from there `../bar` cannot be found.
But in the initial issue where I became aware of this ([straight-shoota/crinja#1](https://github.com/straight-shoota/crinja/issues/1)), the required file was `../value` which seems to resolve to `value.cr` in Crystal stdlib.

**Comment by RX14** [comment]
Seems to me that this is a bug: relative paths should never be looked up in the crystal path.

**Comment by MakeNowJust** [comment]
@RX14 I think so too now because there is no spec about this behavior in [`spec/compiler/crystal_path/crystal_path_spec.cr`](https://github.com/crystal-lang/crystal/blob/fece2f795fb5baeb24ae410be2a99ca277be862f/spec/compiler/crystal_path/crystal_path_spec.cr). And I fixed this in #4765. Could you review it?
